Adding empty string and empty list as falsey values

mmellado opened the pull request:
Adding a brief description of the API and how to use it to the README file to make it easier for users new to unit testing to understand the purpose of each mixin

mirisuzanne commented on README.md:
````diff
@@ -181,3 +181,150 @@ toggles output to the terminal on or off.
+  }
+```
+
+#### `@include output`
````
Maybe combine the docs for `output` and `expect` (and even `assert`) — since these are required together to create a single assertion similar to `assert-true` or `assert-false`?

mirisuzanne commented on README.md:
```diff
@@ -181,3 +181,150 @@ toggles output to the terminal on or off.
+
+#### `@include assert-true()`
+
+Asserts that the output of a test is `true`.
```
I think any non-false value will pass this test...

mirisuzanne commented on README.md:
```diff
@@ -181,3 +181,150 @@ toggles output to the terminal on or off.
+
+#### `@include assert-false()`
+
+Asserts that the output of a test is `false`.
```
`null` also returns falsey for this assertion. Empty strings and empty lists do not, but maybe they should? We could add an issue to fix that.

mmellado commented:
They all make sense. I'll consolidate `assert`, `output` and `expect` into
a single block and update the docs for `assert-true` and `assert-false`. I
do agree that empty string and empty list should be considered falsy. Feel
free to open an issue and I can update this PR with the necessary updates
to accomplish this.

On Mar 2, 2017 7:31 PM, "Miriam Suzanne" <notifications@github.com> wrote:

> *@mirisuzanne* commented on this pull request.
>
> Thanks! This looks great. I left a few comments. Thoughts?
> ------------------------------
>
> In README.md
> <https://github.com/oddbird/true/pull/90#discussion_r103994446>:
>
> > +assert but can have as many as necessary based on your testing needs.
> +
> +#### `@include assert()`
> +
> +Defines a single assert. Use `output()` and `expect()` mixins to define blocks
> +for comparison.
> +
> +```scss
> +  @include test('Your test description') {
> +    @include assert('Your assert description') {
> +      // We will test for an mixin input and output here
> +    }
> +  }
> +```
> +
> +#### `@include output`
>
> Maybe combine the docs for output and expect (and even assert) — since
> these are required together to create a single assertion similar to
> assert-true or assert-false?
> ------------------------------
>
> In README.md
> <https://github.com/oddbird/true/pull/90#discussion_r103994647>:
>
> > +  @include test('Your test description') {
> +    @include assert('Your assert description') {
> +      @include output {
> +        @include the-mixin-to-be-tested();
> +      }
> +
> +      @include expect {
> +        // The expected output of the-mixin-to-be-tested()
> +      }
> +    }
> +  }
> +```
> +
> +#### `@include assert-true()`
> +
> +Asserts that the output of a test is `true`.
>
> I think any non-false value will pass this test...
> ------------------------------
>
> In README.md
> <https://github.com/oddbird/true/pull/90#discussion_r103995253>:
>
> > +Asserts that the output of a test is `true`.
> +
> +- `@param` {`*`} `$assert` - Assert to be tested
> +- `@param` {`String`} `$description` [''] - Assert description
> +
> +```scss
> +  @include test('Your test description') {
> +    $test: sample-function(5);
> +
> +    @include assert-true($test, 'Your assert description');
> +  }
> +```
> +
> +#### `@include assert-false()`
> +
> +Asserts that the output of a test is `false`.
>
> null also returns falsey for this assertion. Empty strings and empty
> lists do not, but maybe they should? We could add an issue to fix that.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/oddbird/true/pull/90#pullrequestreview-24805626>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AASrhYiQO29biMvDw0rZzw-0TGbNDjHfks5rhwsBgaJpZM4MO3ve>
> .
>

mmellado commented:
I've updated the README accordingly and added checks to consider empty string and empty list as falsey values.  Let me know if you'd like to go with a different approach. If in the end it's decided that empty string and empty list should not be considered false I can roll back to update only the README :)

mirisuzanne commented:
Thanks!!

mmellado commented:
My pleasure ;)
